client/src/app/site/base/base-repository.ts

```diff
@@ -75,7 +75,7 @@ export abstract class BaseRepository<V extends BaseViewModel, M extends BaseMode
      * @param update the update that should be created
      * @param viewModel the view model that the update is based on
      */
-    public abstract save(update: M, viewModel: V): Observable<M>;
+    public abstract update(update: M, viewModel: V): Observable<M>;
```

Comment:
shouldnt `M` extend a `BaseModel` and V extend `ViewModel`? Cause "type safety" :-P

Reply:
In the lines above it is already that way